dashboard/db/migrate/20250507130924_add_key_to_skills.rb

```diff
@@ -0,0 +1,6 @@
+class AddKeyToSkills < ActiveRecord::Migration[6.1]
+  def change
+    add_column :skills, :key, :string, null: false
+    add_index :skills, :key, unique: true
```

Comment:
I'm 90% sure we need this (I couldn't get things working without it). If review confirms it's needed, I can extract it into it's own PR.

Reply:
yes, the index seems like a good idea. yes, separate PR is ideal for migrations.